### PR TITLE
Audio: clean up stream data and processing functions in crossover on reset.

### PR DESCRIPTION
**Pull request by eddy1021**
This PR contributes to issue #4488

### CODE REVIEW

File: src/audio/crossover/crossover.c [hunk 1]
```diff
@@ -835,6 +835,11 @@ static int crossover_reset(struct comp_dev *dev)
 
 	crossover_reset_state(cd);
 
+	cd->crossover_process = NULL;
```
Comment (lgirdwood) [inline src/audio/crossover/crossover.c:838]: would these assignments be better in ```crossover_reset_state();```

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

File: src/audio/crossover/crossover.c (hunk 1 above)
Comment (eddy1021) [reply src/audio/crossover/crossover.c:838]: `crossover_reset_state()` will be called in `crossover_setup()`([code](https://github.com/thesofproject/sof/blob/main/src/audio/crossover/crossover.c#L294)) and `crossover_setup()` will be called in `crossover_copy()`([code](https://github.com/thesofproject/sof/blob/main/src/audio/crossover/crossover.c#L648)) and `crossover_prepare()`([code](https://github.com/thesofproject/sof/blob/main/src/audio/crossover/crossover.c#L781)).

In these two paths, IIUC, configs and processing functions should not be cleaned up.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt good to merge ?

**Comment by wszypelt** [comment]
@lgirdwood Not yet, the tests are in the queue, I think the results should be within 2-3 hours and the update of the results on Github should be correct

File: src/audio/crossover/crossover.c [hunk 2]
```diff
@@ -835,6 +835,11 @@ static int crossover_reset(struct comp_dev *dev)
...
+	cd->crossover_split = NULL;
+	cd->config = NULL;
```
Comment (cujomalainey) [inline src/audio/crossover/crossover.c:840]: this is a config param, we should not be clearing it, this is leaking memory, also can you update the config system to use the new model_handler?

Comment (eddy1021) [reply src/audio/crossover/crossover.c:840]: Create a PR to fix it: pull #5713

**Pull request merged**